src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -21,35 +28,202 @@ const ActorInit En_Js_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A890C0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A8910C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A89160.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A891C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A89294.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A89304.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/EnJs_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A895C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/func_80A895F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Js/EnJs_Draw.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x11, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 40, 0, { 0, 0, 0 } },
+};
+s32 D_80A896DC[] = { 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000 };
+
+extern SkeletonHeader D_06005EA0;
+extern AnimationHeader D_0600045C;
+extern AnimationHeader D_0600018C;
+
+void En_Js_SetupAction(EnJs* this, EnJsActionFunc actionfunc) {
+    this->actionfunc = actionfunc;
+}
+
+void EnJs_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnJs* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 36.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06005EA0, &D_0600045C, this->limbDrawTable,
+                     this->transitionDrawTable, 13);
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_0600045C);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    En_Js_SetupAction(this, func_80A89304);
+    this->unk_284 = 0;
+    this->actor.gravity = -1.0f;
+    Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_JSJUTAN, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 0, 0);
+}
+
+void EnJs_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnJs* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+u8 func_80A88F64(EnJs* this, GlobalContext* globalCtx, u16 arg2) {
+    s16 temp;
+    s32 temp_2;
+
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        return 1;
+    } else {
+        this->actor.textId = arg2;
+        temp = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+
+        if (temp >= 0) {
+            temp_2 = temp;
+        } else {
+            temp_2 = -temp;
+        }
+
+        if (temp_2 < 0x1801 && this->actor.xzDistFromLink < 100.0f) {
```

Comment:
<= 1800 should work

Reply:
you should be able to use the ABS macro directly in the conditional, but maybe not